app/controllers/spree/admin/variants_controller_decorator.rb

```diff
@@ -10,12 +10,13 @@ def search
 
   def destroy
     @variant = Spree::Variant.find(params[:id])
-    @variant.delete # This line changed, as well as removal of following conditional
+    @variant.delete_and_refresh_cache
```

Comment:
I'm not a big fan of this but I consider it a good-enough solution given the changes pointed out by @mkllnk in v2 regarding this. #conservative.